### DISCUSSION TITLE
Common steps for all workflows?

### DISCUSSION THREAD

**Discussion by pmeier** [community/community#25901] [Actions]
<p>In most of my runners, the first couple of steps are identical, since they setup a common environment. Is it possible to bundle common steps in one place and use them from other workflows? It is error prone to maintain these steps in every workflow independently.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>That sounds like using the <a href="https://github.blog/changelog/2020-08-07-github-actions-composite-run-steps/">recently introduced composite actions</a> might help. They allow you to combine a bunch of <code>run</code> steps into a reusable action. Note that using other actions in composites is not supported yet, so you’d still have to do things like checkout in each workflow.</p>

**Comment by pmeier** [comment]
<p>When</p>
<blockquote>
<p>using other actions in composites</p>
</blockquote>
<p>is supported, this is exactly what I need. Do you know if there is a timeline on that?</p>

**Comment by airtower-luna** [comment]
<p>Unfortunately not, I don’t have access to Github’s internal planning either. I only know they’re working on it: <a href="https://github.com/actions/runner/issues/646">https://github.com/actions/runner/issues/646</a></p>
